conan/tools/apple/apple.py

```diff
@@ -11,16 +12,10 @@ def is_apple_os(conanfile):
     return str(os_) in ['Macos', 'iOS', 'watchOS', 'tvOS']
 
 
-def to_apple_arch(arch):
+def to_apple_arch(conanfile):
     """converts conan-style architecture into Apple-style arch"""
-    return {'x86': 'i386',
-            'x86_64': 'x86_64',
-            'armv7': 'armv7',
-            'armv8': 'arm64',
-            'armv8_32': 'arm64_32',
-            'armv8.3': 'arm64e',
-            'armv7s': 'armv7s',
-            'armv7k': 'armv7k'}.get(str(arch))
+    arch_ = conanfile.settings.get_safe("arch")
```

Comment:
I am slightly concerned about some recent tools similar to this, as I am wondering what will happen if at some point users want to use it for the build context. They will not be able. So the options would be:
- To expose the underlying dict, something like ``apple_archs(arch)``
- To create a new ``to_apple_arch_build(conanfile)`` (a bit ugly?)

I am fine with this change at this moment, but something to have in mind and maybe discuss.

Reply:
That's a good point! If there's a risk that we may end up with too many small tool functions that have a similar issue, I'd be more inclined to try and sort this out sooner rather than later. `to_apple_arch` has very few uses in Conan Center at the moment, so we may have an opportunity to get it right to cover all cases (although we also want to avoid a repeat of the situations where recipes are using it from the "private" module `apple.apple`)

Reply:
there are some other alternatives, like:
- change context in scope via RAII, e.g. `with context_switch(ctx="build"): do something`
- provide a nice moniker like `if is_apple_arch(conanfile.build)` where `conanfile.build` is `ConanFIle` instance with context switched to build

Reply:
Related https://github.com/conan-io/conan/issues/11926